File: cip/src/formats/epub_parser.cpp
// Copyright (c) 2022 Xu Shaohua <shaohua@biofan.org>. All rights reserved.
// Use of this source is governed by GNU General Public License
// that can be found in the LICENSE file.

#include "formats/epub_parser.h"

#include <QDebug>

#include "formats/epub_reader.h"

bool ParseEpubFile(const QString& filepath) {
  EpubReader reader;
  if (reader.load(filepath)) {
    const auto pages = reader.numPages();
    qDebug() << "page num:" << pages;
    QString text;
    const bool ok = reader.readPage(1, text);
    qDebug() << "ok:" << ok;
    qDebug() << "text:\n" << text;
    return true;
  }

  return false;
}

File: cip/src/parsers/epub_parser.rs
// Copyright (c) 2023 Xu Shaohua <shaohua@biofan.org>. All rights reserved.
// Use of this source is governed by GNU General Public License
// that can be found in the LICENSE file.

use std::path::Path;

use crate::error::Error;
use crate::parsers::cip::CipRecord;

pub fn parse_cip_from_epub<P: AsRef<Path>>(_path: P) -> Result<CipRecord, Error> {
    todo!()
}
